### ISSUE TITLE
Remove deprecated PWM APIs

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#1859]
**_Reported by Anas Nashif:_**

Its been 2 releases and we still carry those deprecated APIs, they need to be removed and drivers need to be adapted.

(Imported from Jira ZEP-2018)

**Comment by nashif** [comment]
**_by Anas Nashif:_**

_Adithya Baglody_ Are you looking into this?

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_Adithya Baglody_ , what's the status of this?  Why did you set the Fix Version/s field to 1.9?

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_Adithya Baglody_ , ah wait, I see that this is blocked by GH-1866 which is now planned for 1.9.  It all makes sense now.  :-)

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

Hi _Adithya Baglody_ .  Anas indicated this past Monday in our review meeting that this is in code review.  That's why I moved the story to the Code Review state for you.  How's the code review coming along?  By when do you expect this to be merged into 1.9?

**Comment by nashif** [comment]
**_by Mark Linkmeyer:_**

_Chandrakala Kempanna_ , please have QA test this.  There is a PWM API test that probably needs to be modified/enhanced.  I'm changing the test flag to testable.

**Comment by nashif** [comment]
**_by Chandrakala Kempanna:_**

_ravi shankar_ Please verify this in the upcoming QA Execution cycle.

**Comment by nashif** [comment]
**_by ravi shankar:_**

tested and verified on quark_d2000_crb board on commit id:c7634a94863a1add8ed1db6f80e34860631cd894
